src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Layout.cs

```diff
@@ -620,15 +714,111 @@ public ViewPosition(GCHandle key, GridPosition value)
 				}
 			}
 
+			// Auto rows should take up the larger of their measured height and their MinHeight
+			for (int i = 0; i < rows.Length; i++)
+			{
+				if (rows[i].Height.IsAuto)
+				{
+					calculatedPixelRows.Span[i] = Math.Max(calculatedPixelRows.Span[i], rows[i].MinHeight);
+				}
+			}
+
+			// Star size: We always measure to set the desired size, but measuring would only be required for the rows calculation when the Star doesn't mean remaining space.
 			var usedHeight = calculatedPixelRows.Span.Sum();
-			double remainingHeight = Math.Max(0, availaibleHeight - usedHeight);
-			var totalStarSizedHeight = GetTotalStarSizedHeight(rows);
-			var defaultStarHeight = remainingHeight / totalStarSizedHeight;
-			var starCalculatedPixelRows = rows
-				.SelectToMemory((r, i) =>
-					r.Height.IsStarSize ?
-						r.Height.StarSize.GetValueOrDefault() * defaultStarHeight :
-						calculatedPixelRows.Span[i]);
+			var initialRemainingHeight = Math.Max(0, availaibleHeight - usedHeight);
+			var initialTotalStarSizedHeight = GetTotalStarSizedHeight(rows);
+
+			var remainingHeight = initialRemainingHeight;
+			var totalStarSizedHeight = initialTotalStarSizedHeight;
+			var unitStarHeight = remainingHeight / totalStarSizedHeight;
+
+			var maxTries = starSizeChildren.Span.Length * 2 + 1;
+			var previousRemainingHeight = remainingHeight;
+
+			/// Try to set the star row heights. If there are no MinHeights or MaxHeights in effect, this should only take a single pass.
+			/// 
+			/// If the actual height consumed is less than was available, this means that at least one row MinHeight kicked in. Conversely,
+			/// if it's greater than what was available, a MaxHeight must have kicked in.
+			/// 
+			/// In either case, we exclude the constrained rows and recalculate for the remaining 'true' star-sized rows. The new unitStarHeight may cause
+			/// more Min/MaxHeights to kick in, so we repeat until the values stabilize.
+			Memory<double> starCalculatedPixelRows = null;
+			for (int i = 0; i <= maxTries; i++)
```

Comment:
If you understand what changes between iterations, you might be able to come up with an algorithm that doesn't need iterations. An example could be to order the columns by their MaxWidth value (ascending) and apply the constraints on that sorted list once.